add reset signal to plots

bryevdv opened the pull request:
- [x] issues: fixes #5071

example:

```py
from bokeh.io import show
from bokeh.layouts import column
from bokeh.models import Button, CustomJS
from bokeh.plotting import figure

p = figure(tools="reset,pan,wheel_zoom,lasso_select")
p.circle(list(range(10)), list(range(10)))

b = Button()
b.js_on_click(CustomJS(args=dict(p=p), code="""
    p.reset.emit()
"""))

show(column(p, b))
```

mattpap commented on bokehjs/src/coffee/models/plots/plot.ts:
```diff
@@ -160,6 +161,8 @@ export interface Plot extends Plot.Attrs {}
 
 export class Plot extends LayoutDOM {
 
+  reset = new Signal0(this, "reset")
```
Only declaration can be here. Definition must be moved to `initialize()`. This is because `initialize()` doesn't work particularly well with this style of member initialization. This will work by default though (in deferred mode), but not when used directly in bokehjs.

mattpap replied:
Sure, I will finish this. I wasn't consistent myself with this. I will need to review how classes are initialized to make sure that there is one, correct way.

bryevdv commented:
I fixed this up, the issue was earlier I had added to `constructor`, not to `initialize`

mattpap commented:
> I fixed this up, the issue was earlier I had added to constructor, not to initialize

Adding to constructor (after super, but TS won't let you add before) is equivalent to your original approach.

bryevdv commented:
> Adding to constructor (after super, but TS won't let you add before) is equivalent to your original approach. 


Weird, not sure why it didn't seem to be working then. In any case I think this addresses your comment, so I will go ahead and merge
